### PR TITLE
osc/pt2pt: fix infinite frag allocation loop

### PR DESCRIPTION
**Pull request by hjelmn**
Signed-off-by: Nathan Hjelm <hjelmn@lanl.gov>
(cherry picked from commit 12b52b2b2cad64f0d98861964286ceb25bdfbd44)
Signed-off-by: Nathan Hjelm <hjelmn@lanl.gov>

### CODE REVIEW

**Comment by ibm-ompi** [comment]
The IBM CI (GNU Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/51b6e8e91673586a73efff49b7b3f3de

**Comment by jjhursey** [comment]
The IBM CI failure is valid, but not related to this PR. @rhc54 it's a failure in PMIx (stack below)
```shell
#0  0x00003fff7c99593c in pmix_server_init ()
   from /home/mpiczar/jenkins/workspace/ompi_public_pr_release_gnu/ompi-install/lib/libopen-rte.so.0
#1  0x00003fff7c296630 in rte_init () from /home/mpiczar/jenkins/workspace/ompi_public_pr_release_gnu/ompi-install/lib/openmpi/mca_ess_hnp.so
#2  0x00003fff7c95121c in orte_init () from /home/mpiczar/jenkins/workspace/ompi_public_pr_release_gnu/ompi-install/lib/libopen-rte.so.0
#3  0x00003fff7c98ed84 in orte_submit_init ()
   from /home/mpiczar/jenkins/workspace/ompi_public_pr_release_gnu/ompi-install/lib/libopen-rte.so.0
#4  0x00000000100012e8 in orterun (argc=7, argv=0x3fffff3a0328) at orterun.c:133
#5  0x0000000010000fc0 in main (argc=7, argv=0x3fffff3a0328) at main.c:13
```

It looks like this is just on the `v3.x` branch (heads up @bwbarrett @hppritcha ). I'm going to disable the IBM testing against the `v3.x` branch until we can get a fix.

**Comment by rhc54** [comment]
@jjhursey I don't know what I can do with that info - is there any way to tell us what failed in that function?

**Comment by jsquyres** [comment]
@jjhursey What's with these warning messages in the IBM CI gist:

```
[c656f6n04:147737] mca_base_component_repository_open: unable to open mca_state_staged_hnp: /home/mpiczar/jenkins/workspace/ompi_public_pr_release_gnu/ompi-install/lib/openmpi/mca_state_staged_hnp.so: undefined symbol: orte_staged_execution (ignored)
[c656f6n04:147737] mca_base_component_repository_open: unable to open mca_state_staged_orted: /home/mpiczar/jenkins/workspace/ompi_public_pr_release_gnu/ompi-install/lib/openmpi/mca_state_staged_orted.so: undefined symbol: orte_staged_execution (ignored)
[c656f6n04:147737] mca_base_component_repository_open: unable to open mca_oob_usock: /home/mpiczar/jenkins/workspace/ompi_public_pr_release_gnu/ompi-install/lib/openmpi/mca_oob_usock.so: undefined symbol: opal_sec (ignored)
```

**Comment by rhc54** [comment]
Those are all removed symbols, and those components should also have been removed. Maybe some mistiming of PR's?

**Comment by jjhursey** [comment]
Humm. I'm investigating. But I'm wondering if Jenkins merged this into v2.x (which has those symbols) instead of v3.x (which doesn't).

**Comment by jjhursey** [comment]
bot:ibm:gnu:retest

**Comment by jjhursey** [comment]
bot:ibm:gnu:retest

**Comment by jjhursey** [comment]
Ok so I found the problem in the IBM CI setup and it's fixed now. We were picking up an old install which was throwing off our release build. This looks clean now. Sorry for the noise... 😞

File: ompi/mca/osc/pt2pt/osc_pt2pt_frag.h [hunk 1]
```diff
@@ -172,6 +172,10 @@ static inline int ompi_osc_pt2pt_frag_alloc (ompi_osc_pt2pt_module_t *module, in
 {
     int ret;
 
+    if (request_len > mca_osc_pt2pt_component.buffer_size) {
+        return OMPI_ERR_OUT_OF_RESOURCE;
+    }
+
     do {
         ret = ompi_osc_pt2pt_frag_alloc (module, target, request_len , buffer, ptr, long_send, buffered);
```
Comment (hjelmn) [inline ompi/mca/osc/pt2pt/osc_pt2pt_frag.h]: @jsquyres Looks like the typo is indeed here. The other pt2pt commit is needed.

Comment (jsquyres) [inline ompi/mca/osc/pt2pt/osc_pt2pt_frag.h]: Ah, ok.  I thought somehow it didn't matter over here.

Will you add c72fb30 to this PR?

**Comment by hjelmn** [comment]
@jsquyres Ready to review.

**Pull request merged**